features/grpc/exporter/src/test/java/org/opennms/features/grpc/exporter/SpogInventoryServiceSyncIT.java

```diff
@@ -0,0 +1,570 @@
+        "classpath:/META-INF/opennms/mockEventIpcManager.xml",
+        "classpath*:/META-INF/opennms/component-dao.xml",
+        "classpath:/META-INF/opennms/applicationContext-mockConfigManager.xml"})
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase(dirtiesContext = false, tempDbClass = MockDatabase.class, reuseDatabase = false)
+public class SpogInventoryServiceSyncIT implements TemporaryDatabaseAware<MockDatabase> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SpogInventoryServiceSyncIT.class);
+    private static final int PORT = 50051;
+    private static final String TENANT_ID = "opennms-prime";
+    private static final String HOST_NAME = "localhost:" + PORT;
+
+    private Server server;
+    private SpogGrpcClient client;
+
+    @Autowired
+    private DatabasePopulator databasePopulator;
+
+    @Autowired
+    private HwEntityDao hwEntityDao;
+
+    @Autowired
+    private NodeDao nodeDao;
+
+    @Autowired
+    private AlarmDao alarmDao;
+
+    @Autowired
+    private MockEventIpcManager eventdIpcMgr;
+
+    private int _id = 1;
+    private long nodeId = 0;
+    private MockDatabase mockDatabase;
+
+    private NmsInventoryServiceSyncImpl grpcServerService = null;
+    private CompletableFuture<Boolean> serverResponseFuture = null;
+
+    private AlarmExporter alarmExporter;
+    private EventsExporter eventsExporter;
+    private InventoryExporter inventoryExporter;
+    private RuntimeInfo runtimeInfo;
+    private EventSubscriptionService eventSubscriptionService;
+    private SpogInventoryService spogInventoryService;
+
+
+    @Before
+    public void setUp() throws Exception {
+        eventdIpcMgr.setEventWriter(mockDatabase);
+        // create data for node
+        populateData();
+        serverResponseFuture = new CompletableFuture<>();
+        grpcServerService = new NmsInventoryServiceSyncImpl(serverResponseFuture);
+        // configure server and client
+        initializeAndStartServer();
+        initializeAndStartClient();
+
+        runtimeInfo = mock(RuntimeInfo.class);
+        this.eventSubscriptionService = mock(EventSubscriptionService.class);
+    }
+
+    private void populateData() {
+        this.databasePopulator.populateDatabase();
+    }
+
+    private void initializeAndStartServer() throws IOException {
+        server = ServerBuilder.forPort(PORT)
+                .addService(grpcServerService)
+                .build()
+                .start();
+    }
+
+    private void initializeAndStartClient() throws SSLException {
+        client = new SpogGrpcClient(HOST_NAME,
+                null,
+                false,
+                new GrpcHeaderInterceptor(TENANT_ID)
+        );
+        client.start();
+    }
+
+    @Test
+    public void testClientRecoveryAfterServerStopAndRestart() throws Exception {
+        // Step 1: Wait for the channel to be ready and set up the initial inventory update
+        waitForChannelToBeReady();
```

Comment:
You shouldn't need channel wait in all test cases, the client should take care of it